src/nvim/msgpack_rpc/remote_ui.c

```diff
@@ -214,16 +213,11 @@ static void remote_ui_mouse_off(UI *ui)
   push_call(ui, "mouse_off", args);
 }
 
-static void remote_ui_insert_mode(UI *ui)
+static void remote_ui_change_mode(UI *ui, int mode)
 {
   Array args = ARRAY_DICT_INIT;
-  push_call(ui, "insert_mode", args);
-}
-
-static void remote_ui_normal_mode(UI *ui)
-{
-  Array args = ARRAY_DICT_INIT;
-  push_call(ui, "normal_mode", args);
+  ADD(args, INTEGER_OBJ(mode));
```

Comment:
I'd prefer if we sent a "normal"/"insert"/"replace" string instead of an integer in this case. For builtin UIs(tui.c) it is ok to use an integer since it has direct access to neovim headers, but remote uis can be written in any language, so it's simpler to use string identifiers.